sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/AndroidBroadcastingCheck.kt

```diff
@@ -0,0 +1,95 @@
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.kotlin.checks
+
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.calls.model.ResolvedCall
+import org.jetbrains.kotlin.resolve.calls.model.ResolvedValueArgument
+import org.sonar.check.Rule
+import org.sonarsource.kotlin.api.CallAbstractCheck
+import org.sonarsource.kotlin.api.FunMatcher
+import org.sonarsource.kotlin.api.isNull
+import org.sonarsource.kotlin.plugin.KotlinFileContext
+
+private const val MESSAGE = "Make sure that broadcasting intents is safe here."
+
+private val MATCHER = FunMatcher(qualifier = "android.content.Context") {
+    withNames(
+        "sendBroadcast",
+        "sendBroadcastAsUser",
+        "sendOrderedBroadcast",
+        "sendOrderedBroadcastAsUser",
+        "sendStickyBroadcast",
+        "sendStickyBroadcastAsUser",
+        "sendStickyOrderedBroadcast",
+        "sendStickyOrderedBroadcastAsUser",
+    )
+}
+
+private val STICKY_BROADCAST_NAMES = setOf(
+    "sendStickyBroadcast",
+    "sendStickyBroadcastAsUser",
+    "sendStickyOrderedBroadcast",
+    "sendStickyOrderedBroadcastAsUser",
+)
+
+@Rule(key = "S5320")
+class AndroidBroadcastingCheck : CallAbstractCheck() {
+
+    override val functionsToVisit = listOf(MATCHER)
+
+
+    override fun visitFunctionCall(
+        callExpression: KtCallExpression,
+        resolvedCall: ResolvedCall<*>,
+        kotlinFileContext: KotlinFileContext
+    ) {
+        val arguments = resolvedCall.valueArgumentsByIndex ?: return
+        val name = resolvedCall.resultingDescriptor.name.asString()
+
+        if (
+            with(kotlinFileContext.bindingContext) {
+                isSendBroadcast(name, arguments)
+                    || isSendBroadcastAsUser(name, arguments)
+                    || isSendOrderedBroadcast(name, arguments)
+                    || isSendOrderedBroadcastAsUser(name, arguments)
+                    || name in STICKY_BROADCAST_NAMES
+            }
+        ) {
+            kotlinFileContext.reportIssue(callExpression.calleeExpression!!, MESSAGE)
+        }
+    }
+
+    private fun BindingContext.isSendOrderedBroadcastAsUser(
+        name: String,
+        argumentsByIndex: List<ResolvedValueArgument>
+    ) = name == "sendOrderedBroadcastAsUser" && (argumentsByIndex.size > 2 && argumentsByIndex[2].isNull(this))
```

Comment:
You could also use `argumentsByIndex.getOrNull(2)` to avoid the explicit range check. Perhaps a matter of taste, do what you think is nicer. If you do decide to change it, obviously same comment for the funs below.

Reply:
Yep both solutions are not perfect you either check size or treat null :( But I'll use getOrNull to be consistent with the other checks you've implemented